dataRetriever/factory/resolverscontainer/sovereignShardResolversContainerFactory.go

```diff
@@ -48,7 +49,7 @@ func (srcf *sovereignShardResolversContainerFactory) generateSovereignHeaderReso
 	}
 
 	identifierHdr := factory.ExtendedHeaderProofTopic + shardC.CommunicationIdentifier(shardC.SelfId())
-	resolverSender, err := srcf.createOneResolverSenderWithSpecifiedNumRequests(identifierHdr, EmptyExcludePeersOnTopic, shardC.SelfId())
+	resolverSender, err := srcf.createOneResolverSenderWithSpecifiedNumRequests(identifierHdr, EmptyExcludePeersOnTopic, core.MainChainShardId)
```

Comment:
this is chainID not shardID

Reply:
We use shard ID. This is the headers pool cache target ID (== MainChainShardID) that we use to resolve requests